docs/timestamps.md

```diff
@@ -0,0 +1,215 @@
+
+```go
+func IsAgeOfParentsValid() bool {
+    // check that parents are not too old
+    if message.timestamp-parent1.timestamp > Delta {
+        return false
+    }
+    if message.timestamp-parent2.timestamp > Delta {
+        return false
+    }
+    
+    // check that parents are not too young
+    if message.timestamp <= parent1.timestamp {
+        return false
+    }
+    if message.timestamp <= parent2.timestamp {
+        return false
+    }
+    
+    return true
+}
+```
+
+### Message timestamp vs transaction timestamp
+Transactions contain a timestamp that is signed by the user when creating the transaction. It is thus different from the timestamp in the message which is created and signed by the node. We require 
+```
+message.timestamp-tw < transaction.timestamp < message.timestamp
+```
+where `tw` defines the maximum allowed difference between both timestamps.
+
+If a node receives a transaction from a user with an invalid timestamp it does not create a message but discards the transaction with a corresponding error message to the user. To prevent a user's local clock differences causing issues the node should offer an API endpoint to retrieve its `SyncedTime` according to the network time. 
+
+### Reattachments
+Reattachments of a transaction are possible during the time window `tw`. Specifically, a transaction can be reattached in a new message as long as the condition `message.timestamp-tw < transaction.timestamp` is fulfilled. If for some reason a transaction is not *picked up* (even after reattchment) and thus being orphaned, the user needs to create a new transaction with a current timestamp. 
+
+### Age of UTXO
+Inputs to a transaction (unspent outputs) inherit their spent time from the transaction timestamp. Similarly, unspent outputs inherit their creation time from the transaction timestamp as well. For a transaction to be considered valid we require
+```
+inputs.timestamp < transaction.timestamp
+```
+In other words, all inputs to a transaction need to have a smaller timestamp than the transaction. In turn, all created unspent outputs will have a greater timestamp than all inputs.
+
+## Consensus
+The timestamp should define the time when the message was created and issued to the Tangle, and this must be enforced to some degree through voting. Specifically, nodes will vote on whether the timestamp was issued within `w` of current local time. This time window is large to account for the network delay. 
+Clearly, in order to have a correct perception of the timestamp quality, **we assume the node is in sync** (see section [Not in Sync](#Not_in_Sync) otherwise).
+Voting on timestamps should not occur for every messsage. Specifically, only for those that arrive around the border of the threshold +-`w`.
```

Comment:
What protects against an attacker sending many messages that will arrive in the time around the threshold +-`w` threshold? Couldn't that lead to a problem, because then quite a lot of voting is needed?

Reply:
That is correct. However, an attacker could also force a lot of voting by spamming double spends. So this does not really open a new attack vector. 

We're currently restructuring a lot and need to update all the specs with the new details. In #880 you can see a new data flow (still subject to changes at the moment). Generally, everything that happens *after* the scheduler is governed by the congestion control algorithm. As such, an attacker can not force unlimited voting (be it through conflicts or timestamps) but is limited by its fair share